packages/loader/container-loader/src/containerContext.ts

```diff
@@ -223,9 +223,6 @@ export class ContainerContext implements IContainerContext {
 
     public setConnectionState(connected: boolean, clientId?: string) {
         const runtime = this.runtime;
-
```

Comment:
I think it's not enough to remove this check. context.connected is a public API and its state will not match events we deliver to runtime. I think this layer should not reach out to this.container.connected (other than in constructor), and instead remember events we sent to runtime (what state we communicated) and return that in connected() getter.
Even better would be to deprecate it, but it takes time (and it's not that straightforward, given runtime checks it in constructor)